modules/SofaSparseSolver/src/SofaSparseSolver/SparseCholeskySolver.cpp

```diff
@@ -108,10 +110,117 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+    
+    perm = new int[A.n ];
+    iperm = new int[A.n ];
+
+    fill_reducing_perm(A , perm, iperm); // compute the fill reducing permutation
```

Comment:
It means you compute the permutation at each time step. I am not sure that it is cheap, so I suggest you to add a mechanism to do it only once.

Reply:
It is not that expensive but still cost some computation. It was easier for SparseLDL that already used symbolic decomposition. It may need some work.